### ISSUE TITLE
Wrong interceptor execution order

### ISSUE THREAD

**Issue by AnotherRegularDude** [bigcommerce/gruf#139]
## Please describe the issue

Your yard documentation for the `Gruf::Interceptors::Context#intercept!` method says that interceptors are used in FIFO order. From your code, it looks like FILO order is used because you use the `pop` method instead of `shift`.

## How to Reproduce

Create two interceptors:

```ruby
class FirstInterceptor < Gruf::Interceptors::ServerInterceptor
  def call
    p 1
    yield
  end
end

class SecondInterceptor < Gruf::Interceptors::ServerInterceptor
  def call
    p 2
    yield
  end
end
```
Then use them, expecting them to be executed in FIFO order

```ruby
Gruf.configure do |config|
  config.interceptors.clear
  config.interceptors.use(FirstInterceptor)
  config.interceptors.use(SecondInterceptor)
end
```

After we make the request we will see `2 1` instead of `1 2`.

## What _should_ happen?

We should see `1 2`

## Anything else we should know?

* gruf (2.11.0)
* grpc (1.41.0)
* grpc-tools (1.41.0)
* MRI 3.0.2
* Linux

**Comment by splittingred** [comment]
Yep, I've confirmed this. This is a good catch - fix incoming.

**Comment by splittingred** [comment]
This has been fixed in 2.12.0 - https://rubygems.org/gems/gruf/versions/2.12.0

**Comment by AnotherRegularDude** [comment]
Thank you!
